pandas/core/internals/managers.py

```diff
@@ -106,7 +107,12 @@ class BlockManager(PandasObject):
 
     def __init__(self, blocks, axes, do_integrity_check=True):
         self.axes = [ensure_index(ax) for ax in axes]
-        self.blocks = tuple(blocks)
+
+        non_native_byteorder = '>' if sys.byteorder == 'little' else '<'
+        self.blocks = tuple(block.astype(block.dtype.newbyteorder('='))
+                            if non_native_byteorder in str(block.dtype)
```

Comment:
does this affect all block types? I would rather have a single check to avoid perf issues for this

Reply:
It won't affect the block types which are from custome pandas dtypes and all non-numeric dtypes like boolean. I stumbled upon the error in this issue, because I read a big endian formatted binary file, so everything is basically big endian. In those cases, a single check would actually suffice. Here I am going for the worst case, in case someone put in some mixed typed (little and big endian) object to convert it into a pandas object. The letter case is very unlikely, so I could just do a single check, if you want.

Reply:
ok just do this slightly differently

```
if is_platform_little_endian():
     self.blocks = .....
else:
         # original code here
    self.blocks = ...
```

trying to avoid *any* perf hit for little endian